components/dashboard/src/components/DropDown2.tsx

```diff
@@ -0,0 +1,201 @@
+                className={
+                    "h-16 bg-gray-100 dark:bg-gray-800 flex items-center px-2 " +
+                    (showDropDown
+                        ? "rounded-t-lg"
+                        : "rounded-lg hover:bg-gray-200 dark:hover:bg-gray-700 cursor-pointer")
+                }
+                onClick={toggleDropDown}
+            >
+                {props.children}
+                <div className="flex-grow" />
+                <div className="mr-2">
+                    <Arrow direction={showDropDown ? "up" : "down"} />
+                </div>
+            </div>
+            {showDropDown && (
+                <>
+                    <div className="absolute w-full top-12 bg-gray-100 dark:bg-gray-800 max-h-72 overflow-auto rounded-b-lg mt-3 z-50 p-2">
+                        {!props.disableSearch && (
+                            <div className="h-12">
+                                <input
+                                    type="text"
+                                    autoFocus
+                                    className={"w-full focus rounded-lg"}
+                                    placeholder={props.searchPlaceholder}
+                                    value={search}
+                                    onChange={(e) => setSearch(e.target.value)}
+                                />
+                            </div>
+                        )}
+                        <ul>
+                            {filteredOptions.length > 0 ? (
+                                filteredOptions.map((element) => {
+                                    let selectionClasses = `dark:bg-gray-800 cursor-pointer`;
+                                    if (element.id === selectedElementTemp) {
+                                        selectionClasses = `bg-gray-200 dark:bg-gray-700 cursor-pointer  focus:outline-none focus:ring-0`;
+                                    }
+                                    if (!element.isSelectable) {
+                                        selectionClasses = ``;
+                                    }
+                                    return (
+                                        <li
+                                            key={element.id}
+                                            id={element.id}
+                                            tabIndex={0}
+                                            className={"h-16 rounded-lg flex items-center px-2 " + selectionClasses}
+                                            onMouseDown={() => {
+                                                if (element.isSelectable) {
+                                                    setFocussedElement(element.id);
+                                                    onSelected(element.id);
+                                                }
+                                            }}
+                                            onMouseOver={() => setFocussedElement(element.id)}
```

Comment:
This part felt a little weird to me, having the mouseover select the option rather than requiring a click or keyboard navigation to select it. It also seems to be making the scroll container do a bit of a jittery scroll as you mouse over options at the edge of what's currently in view.